optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverConfig.java

```diff
@@ -235,6 +236,8 @@ public static SolverConfig createFromXmlReader(Reader reader, ClassLoader classL
     protected Class<? extends ThreadFactory> threadFactoryClass = null;
 
     protected Class<?> solutionClass = null;
+
+    protected DomainAccessType domainAccessType = null;
```

Comment:
Putting in between solutionClass and entityClassList (in xsd, fields, get/set/with methods, etc) seems wierd, because:
- it applies to both
- if a user defines solutionClass, he/she defines entityClassList. But this new field is much rarer.

Proposal A) Put it under entityClassList, because the others are more important (important properties should go first)
Proposal B) Out it above solutionClass because it's a general property.

+1 for A, +0 for B.